components/mod-event/useModEventList.tsx

```diff
@@ -404,32 +462,39 @@ export const useModEventList = (
       return
     }
 
-    const items = new Set<string>()
+    if (!results.data?.pageParams) {
+      addFromEventsToWorkspace(modEvents)
+      setShowWorkspaceConfirmation(null)
+      return
+    }
 
-    modEvents.forEach((event) => {
-      if (showWorkspaceConfirmation === 'subjects') {
-        if (ComAtprotoAdminDefs.isRepoRef(event.subject)) {
-          items.add(event.subject.did)
-        } else if (ComAtprotoRepoStrongRef.isMain(event.subject)) {
-          items.add(event.subject.uri)
-        } else if (ChatBskyConvoDefs.isMessageRef(event.subject)) {
-          items.add(event.subject.did)
-        }
-      } else if (showWorkspaceConfirmation === 'creators') {
-        items.add(event.createdBy)
-      } else if (showWorkspaceConfirmation === 'subject-authors') {
-        if (
-          ComAtprotoAdminDefs.isRepoRef(event.subject) ||
-          ChatBskyConvoDefs.isMessageRef(event.subject)
-        ) {
-          items.add(event.subject.did)
-        } else if (ComAtprotoRepoStrongRef.isMain(event.subject)) {
-          items.add(new AtUri(event.subject.uri).host)
-        }
+    setIsAddingToWorkspace(true)
+    const newAbortController = new AbortController()
+    abortController.current = newAbortController
+
+    try {
+      let cursor = results?.data.pageParams[0] as string | undefined
+      do {
+        // When we just want the dids of the users, no need to do an extra fetch to include repos
+        const nextPage = await modEventsGetter(
+          {
+            pageParam: cursor,
+          },
+          { signal: abortController.current?.signal },
+        )
+        await addFromEventsToWorkspace(nextPage.events)
+        cursor = nextPage.cursor
+        //   if the modal is closed, that means the user decided not to add any more user to workspace
+      } while (cursor && showWorkspaceConfirmation)
+    } catch (e) {
+      if (abortController.current?.signal.reason === 'user-cancelled') {
```

Comment:
You can also use the `aborted` property, which is the more common way of checking if a signal was aborted.

```suggestion
      if (abortController.current?.signal.aborted) {
```